### PR TITLE
CELDEV-883 - Delegate adding Calendar Restriction to Query Builder

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-883

### CODE REVIEW

File: component/src/main/java/com/celements/calendar/engine/CalendarEngineLucene.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/engine/CalendarEngineLucene.java]:
```suggestion
```

File: component/src/main/java/com/celements/calendar/engine/CalendarEngineLucene.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/engine/CalendarEngineLucene.java]:
```suggestion
      LOGGER.debug("getFirst/LastEvent: no Event for cal [{}], first [{}]", cal, first);
```

File: component/src/main/java/com/celements/calendar/search/IEventSearchQuery.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/search/IEventSearchQuery.java]: `@Nullable` missing

File: component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java]: `Preconditions.checkState(boolean, Object)` preferable here

File: component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java]: this can be replaced with `com.celements.search.lucene.LuceneUtils.exactify(List<String>)`.
also already does null checking

**Review by msladek — CHANGES_REQUESTED** [review]

File: component/src/main/java/com/celements/calendar/Calendar.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/Calendar.java]:
```suggestion
  private static final List<String> _NON_EVENT_PROPERTYS = ImmutableList.of("lang",
```

File: component/src/main/java/com/celements/calendar/Calendar.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/Calendar.java]:
```suggestion
      LOGGER.error("Failed to get allowed spaces for [{}]", getDocumentReference(),
```

File: component/src/main/java/com/celements/calendar/Calendar.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/Calendar.java]:
```suggestion
    return ImmutableList.of();
```

File: component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java:92]: This method can be simplified with centralized logic in LuceneUtils, e.g.:
```
List<String> ret = LuceneUtils.exactify(allowedSpaces);
if (ret.isEmpty()) {
  // inexistent space
  ret.add("\".\"");
}
return ret
```

File: component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java]: less readable line breaking

File: component/src/main/java/com/celements/calendar/search/DefaultEventSearchQuery.java
Comment (msladek) [inline component/src/main/java/com/celements/calendar/search/DefaultEventSearchQuery.java:24]:
```suggestion
  private List<String> sortFields = ImmutableList.of();
```

File: component/src/test/java/com/celements/calendar/engine/CalendarEngineLuceneTest.java
Comment (msladek) [inline component/src/test/java/com/celements/calendar/engine/CalendarEngineLuceneTest.java]:
```suggestion
```

**Review by msladek — CHANGES_REQUESTED** [review]

File: component/src/main/java/com/celements/calendar/search/DefaultEventSearchQuery.java
Comment (fpichler) [reply component/src/main/java/com/celements/calendar/search/DefaultEventSearchQuery.java:24]: Collections.emptyList is already immutable. Furthermore it is a singleton.

File: component/src/main/java/com/celements/calendar/Calendar.java
Comment (fpichler) [reply component/src/main/java/com/celements/calendar/Calendar.java]: Collections.emptyList is already immutable. Furthermore it is a singleton.

File: component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java
Comment (fpichler) [reply component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java:92]: this is not the same. Exactify is handling a string and getAllowedSpaces is handling an array of Strings. I did not touch this method and do not see why I should risk to brake it. (not enought direct unit testing available to guard a refactoring)

File: component/src/main/java/com/celements/calendar/Calendar.java
Comment (msladek) [reply component/src/main/java/com/celements/calendar/Calendar.java]: ImmutableList.of is also immutable and a singleton.
Imho it's preferrable for building any immutable array for consistency and since it is more explicit about the immutability with it's naming.

I leave this up to you.

File: component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java
Comment (msladek) [reply component/src/main/java/com/celements/calendar/search/CalendarEventSearchQuery.java:92]: exactify is overloaded for a `List<String>`. you already touched the method, but we can leave it be if you consider it too dangerous.

**Review by msladek — DISMISSED** [review]
see comments before merge.

File: component/src/main/java/com/celements/calendar/Calendar.java
Comment (fpichler) [reply component/src/main/java/com/celements/calendar/Calendar.java]: https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html#of--
> Returns the empty immutable list. This list behaves and performs comparably to Collections.emptyList(), and is preferable mainly for consistency and maintainability of your code.

ok, you have convinced me. ;-)

**Comment by fpichler** [comment]
emptyList change. See comment.

**Review by msladek — APPROVED** [review]

**Pull request merged**
